fmriprep/utils/bids.py

```diff
@@ -193,3 +194,52 @@ def _grp_echos(x):
     subj_data.update({"bold": ses_uids})
 
     return subj_data, layout
+
+
+def write_derivative_description(bids_dir, deriv_dir):
+    from fmriprep import __version__
+
+    desc = {
+        'Name': 'fMRIPrep output',
+        'BIDSVersion': '1.1.1',
+        'PipelineDescription': {
+            'Name': 'fMRIPrep',
+            'Version': __version__,
+            # We could parse the version to reference a commit or tag
+            'CodeURL': 'https://github.com/poldracklab/fmriprep',
+            },
+        'CodeURL': 'https://github.com/poldracklab/fmriprep',
+        'HowToAcknowledge': 'https://github.com/poldracklab/fmriprep',
+        }
+
+    # Keys that can only be set by environment
+    if 'FMRIPREP_DOCKER_TAG' in os.environ:
+        desc['DockerHubContainerTag'] = os.environ['FMRIPREP_DOCKER_TAG']
```

Comment:
Do you suggest to set these env variables when building the docker image?

Reply:
If possible. Or perhaps `fmriprep-docker` would get the actual SHA256 hash and set the environment variable...

These are mostly just stubs indicating there's room to be clever and try to figure out how to get precise container references.